subprojects/plugin-use/src/integTest/groovy/org/gradle/plugin/repository/PluginManagementDslSpec.groovy

```diff
@@ -0,0 +1,248 @@
+                    if(request.requestedPlugin.id.name == 'foo') {
+                        request.useTarget { 
+                            artifact = 'com.acme:foo:+' 
+                        }
+                    }
+                }
+                repositories { 
+                    mavenLocal()
+                }
+            }
+        """
+
+        expect:
+        succeeds 'help'
+    }
+
+
+    def "other blocks can follow the pluginManagement block"() {
+        given:
+        settingsFile << """
+            pluginManagement {}
+            rootProject.name = 'rumpelstiltskin'
+        """
+
+        expect:
+        succeeds 'help'
+    }
+
+
+    def "pluginManagement block is not supported in ProjectScripts"() {
+        given:
+        buildScript """
+            pluginManagement {}
+        """
+
+        when:
+        fails 'help'
+
+        then:
+        failure.assertHasLineNumber(2)
+        failure.assertThatCause(containsString("Only Settings scripts can contain a pluginManagement {} block."))
+        includesLinkToUserguide()
+    }
+
+    def "pluginManagement block is not supported in InitScripts"() {
+        given:
+        def initScript = file "definePluginRepos.gradle"
+        initScript << """
+            pluginManagement {}
+        """
+        args('-I', initScript.absolutePath)
+
+        when:
+        fails 'help'
+
+        then:
+        failure.assertHasLineNumber(2)
+        failure.assertThatCause(containsString("Only Settings scripts can contain a pluginManagement {} block."))
+        includesLinkToUserguide()
+    }
+
+    def "pluginManagement block must come before imperative blocks in the settings.gradle script"() {
+        given:
+        settingsFile << """
+            rootProject.name = 'rumpelstiltskin'
+            pluginManagement {}
+        """
+
+        when:
+        fails 'help'
+
+        then:
+        failure.assertHasLineNumber(3)
+        failure.assertThatCause(containsString("The pluginManagement {} block must appear before any other statements in the script."))
+        includesLinkToUserguide()
+    }
+
+    def "pluginManagement block must come before buildScript blocks in the settings.gradle script"() {
+        given:
+        settingsFile << """
+            buildScript {}
+            pluginManagement {}
+        """
+
+        when:
+        fails 'help'
+
+        then:
+        failure.assertHasLineNumber(3)
+        failure.assertThatCause(containsString("The pluginManagement {} block must appear before any other statements in the script."))
+        includesLinkToUserguide()
+    }
+
+    def "pluginManagement block must be a top-level block (not nested)"() {
+        given:
+        settingsFile << """
+            if (true) {
+                pluginManagement {}
+            }
+        """
+
+        when:
+        fails 'help'
+
+        then:
+        failure.assertThatCause(containsString("Could not find method pluginManagement()"))
```

Comment:
This shouldn't fail actually. `pluginManagement` is a normal method on `Settings`, so it will be available to imperative logic.